modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -69,7 +69,7 @@ public class S3RepositoryPlugin extends Plugin implements RepositoryPlugin, Relo
     }
 
     private final SetOnce<S3Service> service = new SetOnce<>();
-    private final SetOnce<Meter> meter = new SetOnce<>();
+    protected final SetOnce<Meter> meter = new SetOnce<>();
```

Comment:
I am unsure about making this protected. Theoretically it allows to set something in here before `createComponents` is called. This would result in `AlreadySetException` once `createComponents` is called.

May be instead we could keep it private and expose actual meter via:
```
    Meter getMeter() {
        return meter.get();
    }
```
similarly to `org.elasticsearch.repositories.s3.S3RepositoryPlugin#getService`